datatransfer/impl/graphsync/graphsync_impl.go

```diff
@@ -56,16 +60,75 @@ func NewGraphSyncDataTransfer(parent context.Context, host host.Host, gs graphsy
 		dataTransferNetwork,
 		nil,
 		make(map[string]validateType),
+		sync.RWMutex{},
 		make(map[datatransfer.ChannelID]datatransfer.ChannelState),
 		gs,
 		host.ID(),
 		0,
 	}
+	if err := gs.RegisterRequestReceivedHook(true, impl.GsReqRecdHook); err != nil {
+		log.Error(err)
+		return nil
+	}
 	receiver := &graphsyncReceiver{parent, impl}
 	dataTransferNetwork.SetDelegate(receiver)
 	return impl
 }
 
+// GsReqRecdHook is a graphsync.OnRequestReceivedHook hook
+// if an incoming request does not match a previous push request, it returns an error.
+func (impl *graphsyncImpl) GsReqRecdHook(p peer.ID, request graphsync.RequestData) ([]graphsync.ExtensionData, error) {
+	var resp []graphsync.ExtensionData
+	chid, _, err := impl.getChannelIDAndData(request)
+
+	extData := graphsync.ExtensionData{
+		Name: ExtensionDataTransfer,
+		Data: nil,
+	}
+	if err != nil {
```

Comment:
the extension not being present is not an error -- there could be a graphsync request on the graphsync node that simply isn't for DT that is treated like a normal request (i.e. Blockchain sync). But this exposes an API issue with the GS hooks I see. I need to expose change the api two a triple: []graphsync.ExtensionData, didValidate, error -- anyway, not your problem for now.